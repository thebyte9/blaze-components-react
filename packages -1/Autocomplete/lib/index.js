"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
exports.__esModule = true;
var input_1 = __importDefault(require("@blaze-react/input"));
var utils_1 = __importDefault(require("@blaze-react/utils"));
var react_1 = __importStar(require("react"));
var Autocomplete = function (_a) {
    var _b = _a.data, data = _b.data, keys = _b.filterBy, keyValue = _b.keyValue, placeholder = _a.placeholder, selected = _a.selected, uniqueId = _a.utils.uniqueId;
    var _c = react_1.useState(""), inputValue = _c[0], setInputValue = _c[1];
    var _d = react_1.useState(false), showSelect = _d[0], setShowSelect = _d[1];
    var handleChange = function (_a) {
        var value = _a.value;
        setShowSelect(true);
        setInputValue(value);
        filterByValue(value);
    };
    var filterByValue = function (value) {
        return data.filter(function (copy) {
            return !!keys.some(function (key) {
                return copy[key].toLowerCase().includes(value.toLowerCase());
            });
        });
    };
    var handleClick = function (copiedData) {
        setInputValue(copiedData[keyValue]);
        setShowSelect(false);
        selected(copiedData);
    };
    return (react_1["default"].createElement(react_1.Fragment, null,
        react_1["default"].createElement(input_1["default"], { placeholder: placeholder, onChange: handleChange, value: inputValue }),
        showSelect &&
            filterByValue(inputValue).map(function (copiedData, key) { return (react_1["default"].createElement("div", { className: "panel", key: uniqueId(copiedData), onClick: function () { return handleClick(copiedData); }, role: "button", "data-testid": "option-" + (key + 1) }, copiedData[keyValue])); })));
};
Autocomplete.defaultProps = {
    placeholder: "Search",
    selected: function () {
        return;
    }
};
exports["default"] = utils_1["default"](Autocomplete);
