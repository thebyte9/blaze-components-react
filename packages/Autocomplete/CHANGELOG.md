# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.8.0-alpha.11](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.10...v0.8.0-alpha.11) (2021-07-13)

**Note:** Version bump only for package @blaze-react/autocomplete





# [0.8.0-alpha.10](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.9...v0.8.0-alpha.10) (2021-07-13)

**Note:** Version bump only for package @blaze-react/autocomplete





# [0.8.0-alpha.9](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.8...v0.8.0-alpha.9) (2021-07-12)

**Note:** Version bump only for package @blaze-react/autocomplete





# [0.8.0-alpha.8](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.7...v0.8.0-alpha.8) (2021-07-12)

**Note:** Version bump only for package @blaze-react/autocomplete





# [0.8.0-alpha.7](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.6...v0.8.0-alpha.7) (2021-07-12)

**Note:** Version bump only for package @blaze-react/autocomplete





# [0.8.0-alpha.6](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.3...v0.8.0-alpha.6) (2021-07-09)

**Note:** Version bump only for package @blaze-react/autocomplete





# [0.8.0-alpha.3](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.2...v0.8.0-alpha.3) (2021-06-17)

**Note:** Version bump only for package @blaze-react/autocomplete





# [0.8.0-alpha.2](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.1...v0.8.0-alpha.2) (2021-06-17)

**Note:** Version bump only for package @blaze-react/autocomplete





# [0.8.0-alpha.1](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.0...v0.8.0-alpha.1) (2021-06-17)

**Note:** Version bump only for package @blaze-react/autocomplete





# [0.8.0-alpha.0](https://github.com/thebyte9/blaze-components-react/compare/v0.6.2-alpha.0...v0.8.0-alpha.0) (2021-06-17)

**Note:** Version bump only for package @blaze-react/autocomplete





## [0.6.2-alpha.0](https://github.com/thebyte9/blaze-components-react/compare/v0.6.1...v0.6.2-alpha.0) (2021-06-17)

**Note:** Version bump only for package @blaze-react/autocomplete
