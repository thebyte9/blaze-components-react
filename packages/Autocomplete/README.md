## Description

Autocomplete component allows users to quickly find and select one from suggested options.

## Usage

```js
const data = {
  keyValue: "name",
  filterBy: ["name", "description"],
  data: [
    {
      id: 1,
      name: "Laravel",
      description: "Lorem ipsum dolor."
    },
    {
      id: 2,
      name: "React",
      description: "Aliquam tincidunt."
    },
    {
      id: 3,
      name: "Adonis",
      description: "Vestibulum auctor."
    }
  ]
};

<Multiselect data={data} selected={selected => {}} />;
```

## API

##### Autocomplete can receive a number of `props` as follow:

| NAME        |         TYPE          | DEFAULT  |
| :---------- | :-------------------: | :------: |
| data        |        Object         |    {}    |
| placeholder |        String         | 'Search' |
| selected    |       Function        | () => {} |
| children    | single/array of nodes |  empty   |
