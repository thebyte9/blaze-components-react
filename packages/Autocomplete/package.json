{
  "name": "@blaze-react/autocomplete",
  "version": "0.8.0-alpha.52",
  "description": "Autocomplete component allows users to quickly find and select one from suggested options.",
  "main": "./dist/main.js",
  "scripts": {
    "build": "NODE_ENV=production webpack --config webpack.config.js",
    "build-storybook": "build-storybook -c .storybook -o docs-build",
    "lint": "eslint '*/**/*.{ts,tsx}' --ignore-path ../../.eslintignore --fix",
    "test": "jest --passWithNoTests",
    "test:coverage": "jest --coverage --passWithNoTests",
    "pre-publish": "yarn build && yarn lint && yarn test",
    "bundle-analyser": "node ../../node_modules/webpack-bundle-analyzer/lib/bin/analyzer.js ./stats/stats.json"
  },
  "keywords": [],
  "author": "byte9",
  "license": "GPL-3.0",
  "devDependencies": {
    "@babel/cli": "^7.13.14",
    "@babel/core": "^7.13.14",
    "webpack": "^5.28.0",
    "webpack-cli": "^4.6.0"
  },
  "peerDependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "gitHead": "d9cac6fc28dad1d0794718aac09d292019446698"
}
