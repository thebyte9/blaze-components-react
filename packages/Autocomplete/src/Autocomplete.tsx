import React, { Fragment, useState } from 'react';

import Input from '@blaze-react/input';
interface IAutocompleteProps {
  data: {
    keyValue: string;
    filterBy: string[];
    data: Record<string, unknown>[];
  };
  selected?: (...args: unknown[]) => unknown;
  label?: string;
  placeholder?: string;
  disabled?: boolean;
  inputProps?: unknown;
}

const Autocomplete: React.SFC<IAutocompleteProps> = ({
  data: { data, filterBy: keys, keyValue },
  disabled,
  label,
  placeholder,
  selected,
  inputProps,
}) => {
  const [inputValue, setInputValue] = useState<string>('');
  const [showSelect, setShowSelect] = useState<boolean>(false);

  const handleChange = ({ value }: { value: string }): void => {
    setShowSelect(true);
    setInputValue(value);
    filterByValue(value);
  };

  const filterByValue = (value: string): unknown[] =>
    data.filter((copy: unknown) => !!keys.some((key) => copy[key].toLowerCase().includes(value.toLowerCase())));

  const handleClick = (copiedData: unknown): void => {
    setInputValue(copiedData[keyValue]);
    setShowSelect(false);
    selected && selected(copiedData);
  };

  return (
    <Fragment>
      <Input
        {...inputProps}
        placeholder={placeholder}
        label={label}
        onChange={handleChange}
        value={inputValue}
        disabled={disabled}
      />

      {showSelect &&
        filterByValue(inputValue).map(
          (copiedData: unknown, key: number): JSX.Element => (
            <div
              className="panel"
              key={`panel-${key}`}
              onClick={() => handleClick(copiedData)}
              role="button"
              data-testid={`option-${key}`}
            >
              {copiedData[keyValue]}
            </div>
          ),
        )}
    </Fragment>
  );
};

Autocomplete.defaultProps = {
  disabled: false,
  label: '',
  placeholder: 'Search',
  selected: (): void => {
    return;
  },
};

export default Autocomplete;
