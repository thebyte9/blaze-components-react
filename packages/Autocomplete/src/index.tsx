import Input from "@blaze-react/input";
import utils from '@blaze-react/utils';
import React, { Fragment, useState } from "react";

interface IAutocompleteProps {
  data: {
    keyValue: string;
    filterBy: string[];
    data: object[];
  };
  utils: {
    uniqueId: (element: any) => string
  };
  selected: (...args: any[]) => any;
  placeholder?: string;
}
const Autocomplete: React.SFC<IAutocompleteProps> = ({
  data: { data, filterBy: keys, keyValue },
  placeholder,
  selected,
  utils: {
    uniqueId
  }
}) => {
  const [inputValue, setInputValue] = useState<string>('');
  const [showSelect, setShowSelect] = useState<boolean>(false);

  const handleChange = ({ value }: { value: string }): void => {
    setShowSelect(true);
    setInputValue(value);
    filterByValue(value);
  };

  const filterByValue = (value: string): object[] =>
    data.filter((copy: object) =>
      !!keys.some(key => copy[key].toLowerCase().includes(value.toLowerCase()))
    );

  const handleClick = (copiedData: object): void => {
    setInputValue(copiedData[keyValue]);
    setShowSelect(false);
    selected(copiedData);
  };

  return (
    <Fragment>

      <Input placeholder={placeholder} onChange={handleChange} value={inputValue} />

      {showSelect && filterByValue(inputValue).map((copiedData: object): JSX.Element => (
        <div
          className="panel"
          key={uniqueId(copiedData)}
          onClick={() => handleClick(copiedData)}
          role="button">{copiedData[keyValue]}
        </div>
      ))}

    </Fragment>
  );
};
Autocomplete.defaultProps = {
  placeholder: "Search",
  selected: (): void => {
    return;
  }
};
export default utils(Autocomplete);
