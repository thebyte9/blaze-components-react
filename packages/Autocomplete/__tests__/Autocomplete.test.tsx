import React from 'react';
import expect from 'expect';
import { mount } from 'enzyme';
import Autocomplete from '../src';

const data = {
  keyValue: 'name',
  filterBy: ['name'],
  data: [
    {
      id: 1,
      name: 'Lorem'
    },
    {
      id: 2,
      name: 'Ipsum'
    },
    {
      id: 3,
      name: 'Dolor'
    }
  ]
};

const AutocompleteComponent = <Autocomplete data={data} selected={() => {}} />;

describe('Autocomplete component', () => {
  test('should be defined and renders correctly (snapshot)', () => {
    const wrapper = mount(AutocompleteComponent);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  // test('should select first option', () => {
  //   const wrapper = mount(AutocompleteComponent);

  //   wrapper
  //     .find('.form-field--checkbox')
  //     .at(0)
  //     .simulate('click');

  //   expect(
  //     wrapper
  //       .find('div')
  //       .at(0)
  //       .text()
  //   ).toContain('Project1');
  // });

  test('should allow to filter', () => {
    const wrapper = mount(AutocompleteComponent);

    wrapper
      .find('input')
      .at(0)
      .simulate('change', { target: { value: 'Lorem' } });
    expect(
      wrapper
        .find('.panel')
        .at(0)
        .text()
    ).toContain('Lorem');
  });
});
