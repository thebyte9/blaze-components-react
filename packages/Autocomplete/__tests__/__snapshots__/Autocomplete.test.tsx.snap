// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete component should be defined and renders correctly (snapshot) 1`] = `
<WithUtils
  data={
    Object {
      "data": Array [
        Object {
          "description": "PHP framework",
          "id": 1,
          "name": "Laravel",
        },
        Object {
          "description": "Javascript library",
          "id": 2,
          "name": "React",
        },
        Object {
          "description": "Javascript framework",
          "id": 3,
          "name": "Adonis",
        },
      ],
      "filterBy": Array [
        "name",
        "description",
      ],
      "keyValue": "name",
    }
  }
>
  <Autocomplete
    data={
      Object {
        "data": Array [
          Object {
            "description": "PHP framework",
            "id": 1,
            "name": "Laravel",
          },
          Object {
            "description": "Javascript library",
            "id": 2,
            "name": "React",
          },
          Object {
            "description": "Javascript framework",
            "id": 3,
            "name": "Adonis",
          },
        ],
        "filterBy": Array [
          "name",
          "description",
        ],
        "keyValue": "name",
      }
    }
    label=""
    placeholder="Search"
    selected={[Function]}
    utils={
      Object {
        "ErrorMessage": [Function],
        "classNames": [Function],
        "uniqueId": [Function],
      }
    }
  >
    <WithUtils
      label=""
      onChange={[Function]}
      placeholder="Search"
      value=""
    >
      <Input
        disabled={false}
        error={false}
        hideTypeToggle={false}
        label=""
        modifier=""
        onChange={[Function]}
        placeholder="Search"
        required={false}
        type="text"
        utils={
          Object {
            "ErrorMessage": [Function],
            "classNames": [Function],
            "uniqueId": [Function],
          }
        }
        validationMessage="This field is required"
        value=""
      >
        <div
          className="form-field form-field--input  "
        >
          <label
            className=""
          />
          <input
            data-testid="input"
            disabled={false}
            onChange={[Function]}
            placeholder="Search"
            required={false}
            type="text"
            value=""
          />
        </div>
      </Input>
    </WithUtils>
  </Autocomplete>
</WithUtils>
`;
